services/src/main/java/org/exoplatform/officeonline/WOPIService.java

```diff
@@ -792,7 +792,29 @@ public void refreshLock(EditorConfig config,
 
     Node node = nodeByUUID(config.getFileId(), config.getWorkspace());
     lockManager.refreshLock(node, lockId);
+  }
+
+  /**
+   * Delete file.
+   *
+   * @param config the config
+   * @throws FileNotFoundException the file not found exception
+   * @throws RepositoryException the repository exception
+   * @throws LockMismatchException 
+   */
+  public void deleteFile(EditorConfig config) throws FileNotFoundException, RepositoryException, LockMismatchException {
+    Node node = nodeByUUID(config.getFileId(), config.getWorkspace());
+    if (node.isLocked()) {
+      FileLock fileLock = lockManager.getLock(node);
+      if (fileLock != null) {
+        throw new LockMismatchException("File is locked", fileLock.getLockId());
+      }
+      throw new LockMismatchException("File is locked by another service", "");
+    }
 
+    Node parent = node.getParent();
+    node.remove();
```

Comment:
Don't do this until we'll understand it's actually required at this level. Just log a warning it was called.